### PR TITLE
test(compat): add GNU patch compatibility tests

### PR DESCRIPTION
**Pull request by weihanglo**
> Blocked on <https://github.com/bmwill/diffy/pull/55>. Please review from 4587314d5080ff74dd0f0fc9b42bb1e53f0a063a
 
Tests verify diffy produces results compatible with reference tools.
Here we have GNU patch. In the future we'll have `git apply`.

We don't run reference tool locally, as user may not have the tool.
To run it, set `env CI=1` and run the test.

Please see the module level doc for more.

### CODE REVIEW

File: tests/compat/main.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+//! Compatibility tests against reference implementations.
+//!
+//! These tests verify diffy produces results compatible with established tools
+//! Focus is on edge cases and ambiguous behavior,
```
Comment (weihanglo) [inline tests/compat/main.rs:4]: Actually IMHO we should move most tests to compatible tests, if they are testing against some reference tool's behavior.

**Comment by weihanglo** [comment]
Thanks for the cleanup and the review!
